cli-tests/src/upload.rs

```diff
@@ -1,19 +1,38 @@
-use std::fs;
-use std::io::BufReader;
-use std::path::Path;
+use std::{
+    env, fs,
+    io::BufReader,
+    path::{Path, PathBuf},
+};
 
 use api::{CreateRepoRequest, GetQuarantineBulkTestStatusRequest};
 use assert_cmd::Command;
 use assert_matches::assert_matches;
 use context::repo::RepoUrlParts as Repo;
+use escargot::{CargoBuild, CargoRun};
 use junit_mock::JunitMock;
+use lazy_static::lazy_static;
 use tempfile::tempdir;
 use test_utils::{
     mock_git_repo::setup_repo_with_commit,
     mock_server::{spawn_mock_server, RequestPayload},
 };
-use trunk_analytics_cli::codeowners::CodeOwners;
-use trunk_analytics_cli::types::{BundleMeta, FileSetType};
+use trunk_analytics_cli::{
+    codeowners::CodeOwners,
+    types::{BundleMeta, FileSetType},
+};
+
+lazy_static! {
+    static ref CARGO_MANIFEST_DIR: PathBuf = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    static ref CARGO_RUN: CargoRun = CargoBuild::new()
+        .bin("trunk-analytics-cli")
+        .target_dir(CARGO_MANIFEST_DIR.join("../target"))
+        .manifest_path(CARGO_MANIFEST_DIR.join("../cli/Cargo.toml"))
+        .features("force-sentry-env-dev")
```

Comment:
My guess is that every new test file in this crate that wants to invoke the CLI will need to do this? Not that it's a huge lift to copy it into new test files - but is there a Rust-y way to do this once and leverage it across modules?

Also - does enabling the `force-sentry-env-dev` feature in the dev dependencies not take care of this https://github.com/trunk-io/analytics-cli/pull/122/files#diff-48118bb96b547e6c79e43924f9db25f71065d526d458312f2c673458370dd4e3R19?

Reply:
If by modules, you mean modules within `cli-tests/src` then we just need to add a `pub` to this `static ref` and reuse it in within the crate.

If you are thinking about how one might reuse this across crates in the workspace, then we could move this to `test_utils` with `pub` and reuse that here and elsewhere.

Enabling `force-sentry-env-dev` for the dependency only enables it for the `lib.rs`, not for the `main.rs`. Here we are using `escargot` to build the `main.rs` binary with the `force-sentry-env-dev` feature enabled.

Reply:
> If by modules, you mean modules within cli-tests/src then we just need to add a pub to this static ref and reuse it in within the crate.

Yep! this is what I meant. Thanks for the info 🫡